training/src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -58,9 +48,14 @@ public Book create(@RequestBody Book book) {
 
     @DeleteMapping("/{id}")
     public void delete(@PathVariable Long id) {
-        bookRepository.findById(id)
-            .orElseThrow(BookNotFoundException::new);
-        bookRepository.deleteById(id);
+        Optional<Book> bookToDelete = bookRepository.findById(id);
+        if (bookToDelete.isPresent()) {
```

Comment:
If you use `orElseThrow(BookNotFoundException::new)`then you avoid this `if` and also the code will be nicer and easier to read. If the exception is thrown, the flow will be ended, otherwise, the flow will follow and the book will be deleted. Also, by using the `orElseThrow` you won't need to receive an `Optional<Book>` because it assures you that if the book it's not present, an exception will be thrown.